app/models/Person.js

```diff
@@ -134,6 +140,61 @@ personSchema.methods.getType = function() {
     }
 };
 
+personSchema.methods.notifyMissionCompleted = function(mission, next) {
+    var that = this;
+    var INC = 1;
+    var PIONEER_INC = INC;
+    var DIPLOMAT_INC = INC;
+    var EVALUATOR_INC = INC;
+    var GOURMET_INC = INC;
+
+    var pioneerInc = 0;
+    var diplomatInc = 0;
+    var evaluatorInc = 0;
+    var gourmetInc = 0;
+
+    var missionType = mission.getType();
+
+    if(missionType === 'AddLocationMission' ||
+        missionType === 'WhatOptionsMission'){
+        pioneerInc += PIONEER_INC;
+    }
+
+    if(mission.isFirstOfType &&
+        (missionType === 'VisitBonusMission' ||
+            missionType === 'HasOptionsMission' ||
+            missionType === 'WantVeganMission' ||
+            missionType === 'WhatOptionsMission' ||
+            missionType === 'RateOptionsMission' ||
+            missionType === 'OfferQualityMission' ||
+            missionType === 'EffortValueMission')) {
+        pioneerInc += PIONEER_INC;
+    }
+
+    if(missionType === 'HasOptionsMission' ||
+        missionType === 'WantVeganMission' ||
+        missionType === 'GiveFeedbackMission'){
+        diplomatInc = DIPLOMAT_INC;
+    }
+
+    if(missionType === 'RateOptionsMission' ||
+        missionType === 'OfferQualityMission' ||
+        missionType === 'EffortValueMission'){
+        evaluatorInc = EVALUATOR_INC;
+    }
+
+    if(missionType === 'VisitBonusMission' ||
+        missionType === 'BuyOptionsMission'){
+        gourmetInc = GOURMET_INC;
+    }
+    that.attributes.pioneer += pioneerInc;
+    that.attributes.diplomat += diplomatInc;
+    that.attributes.evaluator += evaluatorInc;
+    that.attributes.gourmet += gourmetInc;
+    that.updatedAt = Date.now();
```

Comment:
Would be a good idea to have an updatedAt property on Person. But at the moment that's not the case, so this can be removed.